dev/releases/utils.py

```diff
@@ -152,6 +152,38 @@ def check_whether_github_release_exists(tag):
             return True
     return False
 
+# lightweight vs annotated
+# https://stackoverflow.com/questions/40479712/how-can-i-tell-if-a-given-git-tag-is-annotated-or-lightweight#40499437
+def is_annotated_git_tag(tag):
+    is_annotated = subprocess.run(["git", "for-each-ref", "refs/tags/" + tag],
+                                  check=True, capture_output=True, text=True)
+    if "tag" == is_annotated.stdout.split()[1]:
+        return True
+    if "commit" == is_annotated.stdout.split()[1]:
+        return False
+    error(f"could not check whether tag {tag} is annotated")
```

Comment:
Why not just use this, and get rid of `does_git_tag_exist`?
```suggestion
def is_annotated_git_tag(tag):
    res = subprocess.run(["git", "for-each-ref", "refs/tags/" + tag],
                         capture_output=True, text=True)
    return res.returncode <> 0 and "tag" == res.stdout.split()[1]:
```

Reply:
I'm not sure whether `git for-each-ref` is a plumbing tool, so just in case its output changes I'd like to check whether we found something different from `"tag"` or `"commit"` in our `is_annotated.stdout.split()[1]`. But I'll add the simplification of getting rid of `does_git_tag_exist()`

Reply:
Because I think that if the output of `git for-each-ref` would change, then this would be really confusing to debug.

Reply:
git for-each-ref is explicitly a plumbing command, see https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git.html

Reply:
Aaah, that is where I find that information. Thanks! I have been looking at the man page of `git for-each-ref` and couldn't find anything!

Reply:
Okay, I was just guessing which commands would be plumbing and which would be porcelain. No more guessing!